crates/polars-arrow/src/datatypes/mod.rs

```diff
@@ -301,6 +301,8 @@ impl From<arrow_schema::DataType> for ArrowDataType {
             DataType::RunEndEncoded(_, _) => {
                 panic!("Run-end encoding not supported by polars_arrow")
             },
+            #[allow(unreachable_patterns)]
+            _ => unimplemented!("needs arrow-schema >= v51.0.0"),
```

Comment:
just a small suggestion

```suggestion
            // This ensures that it doesn't fail to compile when new variants are added to Arrow
            #[allow(unreachable_patterns)]
            dtype => unimplemented!("unsupported datatype: {dtype}")
```